src/foundry/client/data/documents/combat.d.ts

```diff
@@ -1,198 +1,210 @@
      * Update the Combat turn order to maintain the same combatant as the current turn.
-     * @param id    - The combatant ID for which to set initiative
-     * @param value - A specific initiative value to set
+     * @param id      - The combatant ID for which to set initiative
+     * @param value   - A specific initiative value to set
      */
     setInitiative(id: string, value: number): Promise<void>;
 
-    /** Return the Array of combatants sorted into initiative order, breaking ties alphabetically by name. */
-    setupTurns(): this["turns"];
+    /**
+     * Return the Array of combatants sorted into initiative order, breaking ties alphabetically by name.
+     */
+    setupTurns(): Combatant[];
 
-    /** Begin the combat encounter, advancing to round 1 and turn 1 */
-    startCombat(): Promise<this | undefined>;
+    /**
+     * Update active effect durations for all actors present in this Combat encounter.
+     */
+    updateEffectDurations(): void;
+
+    /**
+     * Loads the registered Combat Theme (if any) and plays the requested type of sound.
+     * If multiple exist for that type, one is chosen at random.
+     * @param type  - One of [ "startEncounter", "nextUp", "yourTurn" ]
+     * @internal
+     */
+    protected _playCombatSound(type: string): void;
 
     /**
      * Define how the array of Combatants is sorted in the displayed list of the tracker.
      * This method can be overridden by a system or module which needs to display combatants in an alternative order.
-     * By default sort by initiative, next falling back to name, lastly tie-breaking by combatant id.
+     * The default sorting rules sort in descending order of initiative using combatant IDs for tiebreakers.
+     * @param a   - Some combatant
+     * @param b   - Some other combatant
      * @internal
      */
-    protected _sortCombatants(
-      a: InstanceType<ConfiguredDocumentClass<typeof Combatant>>,
-      b: InstanceType<ConfiguredDocumentClass<typeof Combatant>>
-    ): number;
+    protected _sortCombatants(a: Combatant, b: Combatant): number;
 
-    protected override _onCreate(
-      data: this["data"]["_source"],
-      options: DocumentModificationOptions,
-      userId: string
-    ): void;
+    /** @internal */
+    protected _onCreate(data: BaseCombat.ConstructorData, options: DocumentModificationOptions, userId: string): void;
 
-    protected override _onUpdate(
-      changed: DeepPartial<this["data"]["_source"]>,
+    /** @internal */
+    protected _onUpdate(
+      data: DeepPartial<BaseCombat.UpdateData>,
       options: DocumentModificationOptions,
       userId: string
     ): void;
 
-    protected override _onDelete(options: DocumentModificationOptions, userId: string): void;
+    /** @internal */
+    protected _onDelete(options: DocumentModificationOptions, userId: string): void;
 
-    protected override _onCreateEmbeddedDocuments(
+    /** @internal */
+    protected _onCreateEmbeddedDocuments(
```

Comment:
@kmoschcau I'm not entirely sure how to type these embedded document functions.

I am getting this error for `_onCreateEmbeddedDocuments`:

> Property '_onCreateEmbeddedDocuments' in type 'Combat' is not assignable to the same property in base type 'BaseCombat<Any | null> & ClientDocumentMixin<BaseCombat<Any | null>>'.
  Type '(type: string, documents: BaseCombatant<Any | null>[], result: InnerInitializedType<Schema>[], options: DocumentModificationOptions, userId: string) => void' is not assignable to type '(embeddedName: string, documents: Document<any, any, null>[], result: Record<string, unknown>[], options: DocumentModificationOptions, userId: string) => void'.
    Types of parameters 'documents' and 'documents' are incompatible.
      Type 'Document<any, any, null>[]' is not assignable to type 'BaseCombatant<Any | null>[]'.
        Type 'Document<any, any, null>' is missing the following properties from type 'BaseCombatant<Any | null>': _id, actorId, tokenId, sceneId, and 6 more.ts(2416)

I get similar errors for `_onUpdateEmbeddedDocuments` and `_onDeleteEmbeddedDocuments`

Reply:
I am not too sure about that either without looking further into it. I know embedded collections were always a pain.

Reply:
Sounds good to me. It does look like `client-document.d.ts` has not been migrated yet, so that may play a part in resolving errors.